src/common/io.py

```diff
@@ -1146,26 +1147,26 @@ def __init__(self, fname, delayed_trajectory_loading = True):
             fname --
                 Name of file.
         """
-        self._fname = fname
-        self._is_filestream = isinstance(fname, io.IOBase)
-        
+        if isinstance(fname, io.IOBase):
+            if hasattr(fname, "name") and os.path.isfile(fname.name):
+                self._fname = fname.name
+            else:
+                raise JIOError("Not supported file format, needs to be a filename or a stream based on a file.")
```

Comment:
Why is this limitation required?

Reply:
The underlying Cython implemention requires a file in order to be efficient so for now, we require this.